.github/workflows/test-integration.yml

```diff
@@ -0,0 +1,126 @@
+name: test-integration
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 2 * * *" # Run nightly at 2 AM UTC
+  push:
+    branches:
+        - kh-integration-testing    
+    
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  wasm-packages-build:
+    runs-on: ubuntu-latest
+    env:
+      CACHED_PATH: /tmp/nix-cache
+
+    steps:
+      - name: Install Yarn
+        run: npm install -g yarn
+
+      - name: Checkout noir sources
+        uses: actions/checkout@v4
+
+      - name: Checkout acvm sources
+        uses: actions/checkout@v3 # v3 is needed here otherwise this fails in local execution
+        with:
+          repository: noir-lang/acvm
+          path: acvm
+
+      - name: Setup Nix
+        uses: cachix/install-nix-action@v22
+        with:
+          nix_path: nixpkgs=channel:nixos-23.05
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: cachix/cachix-action@v12
+        with:
+          name: barretenberg
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+
+      - name: Restore nix store cache
+        if: env.ACT != 'true'
```

Comment:
What is `env.ACT` and why do we need it?

Reply:
https://github.com/nektos/act allow to simulate github actions locally

Reply:
I don't think this should be introduced in this PR because its orthogonal to nightly integration testing and most of our other actions do not have this

Reply:
It's not destructive though... it's not affecting actual run.